### PR TITLE
Enforce maximum column limits for classes in SQLite

### PR DESCRIPTION
**Pull request by akshay05n**
**Overview:**
This pull request addresses the issue of exceeding maximum column limits for classes in SQLite when running the `SELECT *` query. Currently, querying more columns is possible than allowed, which can lead to failures.

**Changes Made:**
Introduced schema import level checks to verify that no class exceeds the maximum column limit(increased from 2000 to 2200) across all tables.

### CODE REVIEW

**Review by ColinKerr — APPROVED** [review]

File: iModelCore/ECDb/ECDb/SchemaManagerDispatcher.cpp [hunk 1]
```diff
@@ -1627,11 +1628,15 @@ BentleyStatus MainSchemaManager::CanCreateOrUpdateRequiredTables() const
             [co].[IsVirtual] = 0
             AND [cm].[MapStrategy] <> 3
         GROUP BY [cl].[Id], tb.Id
-        HAVING COUNT (*) >= %d;)sql", maxColumns);
+        HAVING COUNT (*) > %d;)sql", maxColumns);
```
Comment (khanaffan) [inline iModelCore/ECDb/ECDb/SchemaManagerDispatcher.cpp:1631]: use SQL parameters and not sprintf() and then use stmt.bindint()

File: iModelCore/ECDb/ECDb/SchemaManagerDispatcher.cpp [hunk 2]
```diff
@@ -1641,6 +1646,60 @@ BentleyStatus MainSchemaManager::CanCreateOrUpdateRequiredTables() const
...
+        SELECT
+            [pm].[ClassId] [ClassId],
+            COUNT (*) + 2 [ColumnCount]
+        FROM
+            [ec_PropertyMap] [pm]
+            JOIN [ec_PropertyPath] [pp] ON [pp].[Id] = [pm].[PropertyPathId]
+        WHERE
+            [pp].[AccessString] NOT IN ('ECInstanceId', 'ECClassId')
+        GROUP BY [pm].[ClassId]
+        HAVING [ColumnCount] > %d)x", selectWildCardLimit).GetUtf8CP();
```
Comment (khanaffan) [inline iModelCore/ECDb/ECDb/SchemaManagerDispatcher.cpp:1666]: use sql parameter and bind the limit using stmt.bindInt()

**Review by khanaffan — CHANGES_REQUESTED** [review]

File: iModelCore/BeSQLite/SQLite/sqlite3.c [hunk 3]
```diff
@@ -13732,7 +13732,7 @@ struct fts5_api {
 ** dozen values in any of the other situations described above.
 */
 #ifndef SQLITE_MAX_COLUMN
-# define SQLITE_MAX_COLUMN 2000
```
[the remark this answers is not in the document]
Comment (akshay05n) [reply iModelCore/BeSQLite/SQLite/sqlite3.c]: Added `SQLITE_MAX_COLUMN` in `bentley-sqlite.c` file

File: iModelCore/ECDb/ECDb/SchemaManagerDispatcher.cpp (hunk 1 above)
Comment (akshay05n) [reply iModelCore/ECDb/ECDb/SchemaManagerDispatcher.cpp]: Updated code to use SQL parameters and `stmt.BindInt()`

File: iModelCore/ECDb/ECDb/SchemaManagerDispatcher.cpp (hunk 2 above)
Comment (akshay05n) [reply iModelCore/ECDb/ECDb/SchemaManagerDispatcher.cpp]: Updated code to use SQL parameters and `stmt.BindInt()`

**Review by khanaffan — APPROVED** [review]

**Pull request merged**